core/jobs/batch_jobs/question_migration_jobs.py

```diff
@@ -0,0 +1,298 @@
+
+        Returns:
+            PCollection. A PCollection of results from the question
+            migration.
+        """
+        all_question_models = (
+            self.pipeline
+            | 'Get all non-deleted question models' >> (
+                ndb_io.GetModels(question_models.QuestionModel.get_all()))
+            # Pylint disable is needed becasue pylint is not able to correclty
+            # detect that the value is passed through the pipe.
+            | 'Add question keys and extract version' >> beam.Map( # pylint: disable=no-value-for-parameter
+                lambda model: (model.id, model.version)
+            )
+        )
+
+        question_summary_models = (
+            self.pipeline
+            | 'Get all non-deleted question summary models' >> (
+                ndb_io.GetModels(
+                    question_models.QuestionSummaryModel.get_all())
+                )
+            # Pylint disable is needed because pylint is not able to correctly
+            # detect that the value is passed through the pipe.
+            | 'Add question summary keys' >> beam.WithKeys( # pylint: disable=no-value-for-parameter
+                lambda question_summary_model: question_summary_model.id)
+        )
+
+        question_objects = (
+            {
+                'question': all_question_models,
+                'question_summary_model': question_summary_models,
+            }
+            | 'Merge objects' >> beam.CoGroupByKey()
+            | 'Get rid of ID' >> beam.Values() # pylint: disable=no-value-for-parameter
+            | 'Reorganize the objects' >> beam.Map(lambda objects: {
+                    'question': objects['question'][0],
+                    'question_summary_model': objects[
+                    'question_summary_model'][0],
+                })
+        )
+
+        all_updated_question_summary_results = (
+            question_objects
+            | 'Update question summary models' >> beam.Map(
+                lambda question_objects: self._update_question_summary(
+                    question_objects['question'],
+                    question_objects['question_summary_model']
+            ))
+        )
+
+        updated_question_summary_results = (
+            all_updated_question_summary_results
+            | 'Generates results' >> (
+                job_result_transforms.ResultsToJobRunResults(
+                    'QUESTION SUMMARY PROCESSED'))
+        )
+        updated_question_summary = (
+            all_updated_question_summary_results
+            | 'Filter oks' >> beam.Filter(
+                lambda result_item: result_item.is_ok())
+            | 'Unwrap ok' >> beam.Map(
+                lambda result_item: result_item.unwrap())
+        )
+
+        question_summary_models_to_put = (
+            updated_question_summary
+            | 'Generate question summary models to put' >> beam.MapTuple(
+                self._populate_question_summary_models
```

Comment:
Optional* Is there any benefit of having a separate update and populate function considering both has the model as input? (not sure!)

Can we just have _update_and_validate_sumarry_model? 

```
def _update_and_validate_sumarry_model(model, version):
    - convert model to object
    - update version number
    - validate
    - convert to model
    - return model
```

Reply:
Yes makes sense. Done